.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: Java CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

Comment:
I think since it's uploading it should strictly be when merging into main imo not just making pull_requests